extensions/amp-ad-network-dianomi-impl/0.1/amp-ad-network-dianomi-impl.js

```diff
@@ -1,20 +1,90 @@
+import {CONSENT_POLICY_STATE} from '#core/constants/consent-state';
+
 import {userAssert} from '../../../src/log';
 import {AmpA4A} from '../../amp-a4a/0.1/amp-a4a';
 
 const REQUEST_PARAM_ID = new RegExp(/^.[0-9]*$/);
 
+/**
+ * Dianomi ad types
+ * @type {Array<string>} TYPES
+ * @private
+ */
+const TYPES = ['smartads', 'recirc', 'context'];
+
+/**
+ * Dianomi environments
+ * @type {object} ENVIRONMENTS
+ * @private
+ */
+const ENVIRONMENTS = {
+  dev: 'dev',
+  live: 'www',
+};
+
 export class AmpAdNetworkDianomiImpl extends AmpA4A {
   /** @override */
-  getAdUrl() {
+  getAdUrl(consentTuple) {
+    let consentString = '';
+    const consentVars = [
+      'consentState',
+      'consentString',
+      'consentStringType',
+      'gdprApplies',
+      'additionalConsent',
+    ];
+    if (
+      consentTuple &&
+      consentTuple.consentState === CONSENT_POLICY_STATE.UNKNOWN
+    ) {
+      return '';
+    }
+    if (consentTuple) {
+      consentString = consentVars.reduce((acc, currentValue) => {
+        if (consentTuple[currentValue]) {
+          acc = `${acc}&${currentValue}=${consentTuple[currentValue]}`;
```

Comment:
I think our `addParamsToUrl` helper in `src/url.js` might make this easier for you.

Reply:
I have used `addParamsToUrl` rather than using my function.